packages/swayze/lib/src/core/controller/selection/selection_controller.dart

```diff
@@ -76,6 +78,20 @@ class SwayzeSelectionController extends Listenable implements ControllerBase {
     userSelections.value = stateUpdate(userSelections.value);
   }
 
+  /// Check if the index [headerIndex] is covered by any header selection.
+  bool isHeaderSelected(int headerIndex, Axis axis) {
+    final selections =
```

Comment:
nit: feel free to ignore. I feel like it's more readable in a single `any` loop:

```dart
userSelectionState.selections.any((selection) => 
  selection is HeaderUserSelectionModel &&
  selection.axis == axis &&
  selection.contains(headerIndex),
);
```